node/src/proxy_server/mod.rs

```diff
@@ -1020,6 +1197,35 @@ impl StreamKeyFactory for StreamKeyFactoryReal {
     }
 }
 
+#[derive(Clone, Debug, Eq, PartialEq)]
+struct DNSFailureRetry {
+    unsuccessful_request: ClientRequestPayload_0v1,
+    retries_left: usize,
+}
+
+#[derive(Clone, Debug, Eq, PartialEq)]
+struct Hostname {
+    hostname: String,
+}
+
+impl Hostname {
+    #[allow(dead_code)]
+    fn new(raw_url: &str) -> Self {
+        let regex = Regex::new(
+            r"^((http[s]?|ftp):/)?/?([^:/\s]+)((/\w+)*/)([\w\-.]+[^#?\s]+)(.*)?(#[\w\-]+)?$",
+        )
+        .expect("Bad Regex");
+        let hostname = match regex.captures(raw_url) {
+            None => raw_url.to_string(),
```

Comment:
Returning the entire URL if it doesn't match seems dangerous. Could you have another regex that you try if this one doesn't match to make sure it's an IP address? You might need _two_ more regexes to be IPv6 compatible.

Reply:
I've added a new card for this issue. and added your comments.
https://github.com/MASQ-Project/MASQ-Node-issues/issues/743